src/Microsoft.IdentityModel.Tokens/InMemoryCryptoProviderCache.cs

```diff
@@ -105,12 +92,12 @@ protected override string GetCacheKey(SignatureProvider signatureProvider)
         /// </summary>
         /// <param name="securityKey">the key that is used to by the crypto provider.</param>
         /// <param name="algorithm">the algorithm that is used by the crypto provider.</param>
-        /// <param name="typeofProvider">the typeof the crypto provider obtained by calling object.GetType().</param>
+        /// <param name="typeofProvider">the type of the crypto provider obtained by calling object.GetType().</param>
         /// <exception cref="ArgumentNullException">if securityKey is null.</exception>
         /// <exception cref="ArgumentNullException">if algorithm is null or empty string.</exception>
         /// <exception cref="ArgumentNullException">if typeofProvider is null or empty string.</exception>
         /// <returns>the cache key to use for finding a crypto provider.</returns>
-        protected override string GetCacheKey(SecurityKey securityKey, string algorithm, string typeofProvider)
+        protected internal override string GetCacheKey(SecurityKey securityKey, string algorithm, string typeofProvider)
```

Comment:
changing the visibility to 'internal' is a breaking change.

Reply:
From MS Docs: "A `protected internal` member of a base class is accessible from any type within its containing assembly. It is also accessible in a derived class located in another assembly only if the access occurs through a variable of the derived class type."

So `protected internal` still allows the same level of access from outside of the assembly as `protected` does. And it also allows our tests to call the methods.